checker.c

```diff
@@ -1,21 +1,53 @@
 #include <stdio.h>
 #include <assert.h>
+int TempAlerter = 0;
+int SOCAlerter = 0;
+int ChargeRateAlerter = 0;
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
+ void CheckIfTempExceedsRange(float temperature)
+  {
+      if(temperature < 0 || temperature > 45)
+      {
+          printf("%fTemperature out of range!\n",temperature);
+          TempAlerter = TempAlerter+1;
+      }
   }
-  return 1;
+ 
+  void CheckIfSOCExceedsRange(float soc)
+  {
+      if(soc < 20 || soc > 80)
+      {
+          printf("%fState of Charge out of range!\n",soc);
+          SOCAlerter = SOCAlerter+1;
+      }
+  }
+  
+   void CheckIfChargeRateExceedsRange(float chargeRate)
+  {
+      if(chargeRate > 0.8)
+      {
+          printf("%fCharge Rate out of range!\n",chargeRate);
+          ChargeRateAlerter = ChargeRateAlerter+1;
+      }
+  }
+
+void batteryIsOk(float temperature, float soc, float chargeRate) 
+{
+    CheckIfTempExceedsRange(temperature);
+    CheckIfSOCExceedsRange(soc);
+    CheckIfChargeRateExceedsRange(chargeRate);
 }
+ 
+ 
 
 int main() {
-  assert(batteryIsOk(25, 70, 0.7));
```

Comment:
Try to cover boundary conditions, out of range values and in range values, for example test values for temperature could be -1, 0, 25, 45, 50. Same can be applied to other parameters too.

Reply:
corrected and updated